tools/azure-sdk-tools/ci_tools/conda/CondaConfiguration.py

```diff
@@ -0,0 +1,196 @@
+#     - package: azure-core
+#       version: 1.24.0
+#     - package: azure-core
+#       version: 1.24.0
+#   - name: azure-storage
+#     common_root: azure
+#     in_batch: ${{ parameters.release_azure_storage }}
+#     checkout:
+#     - package: azure-storage-blob
+#       checkout_path: sdk/storage
+#       version: 12.12.0
+#     - package: azure-storage-queue
+#       checkout_path: sdk/storage
+#       version: 12.3.0
+#     - package: azure-storage-file-share
+#       checkout_path: sdk/storage
+#       version: 12.8.0
+#     - package: azure-storage-file-datalake
+#       checkout_path: sdk/storage
+#       version: 12.7.0
+
+def get_pypi_page(package: str, version: str) -> bs4.BeautifulSoup:
+    url = f"https://pypi.org/project/{package}/{version}/"
+
+    try:
+        r = http.request("GET", url)
+    except Exception as e:
+        raise RuntimeError(f"We must get this the data {url} to continue proper assembly.")
+
+    return bs4.BeautifulSoup(r.data.decode("utf-8"), "html.parser")
+
+
+def get_package_sdist_url(package: str, version: str) -> str:
+    soup = get_pypi_page(package, version)
+
+    # source distribution always first on the page
+    try:
+        target_zip = soup.select("div.file__card a")[0]["href"]
+    except Exception as e:
+        print(f"Cant get data from {package} and {version}")
+        raise e
+    filename = os.path.basename(target_zip)
+
+    if filename.endswith("tar.gz") or filename.endswith(".zip"):
+        return target_zip
+    else:
+        raise ValueError(f"Unable to find a source distribution for {package}@{version}.")
+
+
+class CheckoutConfiguration:
+    def __init__(self, raw_json: dict):
+        # we should always have a package name
+
+        if "package" in raw_json:
+            self.package = raw_json["package"]
+        else:
+            raise ValueError("A checkout configuration MUST have a package name defined in key 'package'.")
+
+        if "checkout_path" in raw_json:
+            self.checkout_path = raw_json["checkout_path"]
+        else:
+            self.checkout_path = None
+
+        if "version" in raw_json:
+            self.version = raw_json["version"]
+        else:
+            self.version = None
+
+        if "download_uri" in raw_json:
+            self.download_uri = raw_json["download_uri"]
+        else:
+            self.download_uri = None
+
+        if "version" in raw_json and self.checkout_path is None:
+            self.download_uri = get_package_sdist_url(self.package, self.version)
+
+        if not self.checkout_path and not self.download_uri:
+            raise ValueError(
+                "When defining a checkout configuration, one must either have a valid PyPI download url"
+                " (download_uri) or a path and version in the repo (checkout_path, version)."
+            )
+
+    def __str__(self) -> str:
+        if self.download_uri:
+            return f"""- {self.package} downloaded from pypi
+  {self.download_uri}"""
```

Comment:
nit:
```suggestion
            return f"""- {self.package} downloaded from pypi {self.download_uri}"""
```

Reply:
Ah this was intentional. Doing it like this results in a nice even:

```
====================================
"azure-storage" generated from:
- azure-storage-blob downloaded from pypi
  https://files.pythonhosted.org/packages/bd/5f/64a471e09f064b3b3a53529ecd9ed8facfebfafff3dad7ee9350f3a00a30/azure-storage-blob-12.17.0.zip
- azure-storage-queue downloaded from pypi
  https://files.pythonhosted.org/packages/69/11/81e41c65182ca2e294efbf42f3b613c18ced546116ce1162846ffe3728ac/azure-storage-queue-12.6.0.zip
- azure-storage-file-share downloaded from pypi
  https://files.pythonhosted.org/packages/68/97/c72fcbd17d7bed7b71ca9bde0d137e1a64ae4cd4952aab557d2766993b11/azure-storage-file-share-12.12.0.zip
- azure-storage-file-datalake downloaded from pypi
  https://files.pythonhosted.org/packages/7f/0e/c367977fe7e45f802698e9a4ecea3bc01f70fadd99d17f754989d2044e2a/azure-storage-file-datalake-12.11.0.zip
====================================
```

Reply:
Look how easy that is to read!